import java.util.InputMismatchException;
import java.util.Scanner;

public class SafeBank {
    public static void main(String[] args) {
        Scanner sc= new Scanner(System.in);

            double saldo = 1000.00;

            System.out.println("\n===== Bem vindo ao SafeBank =====");
            System.out.println("Saldo disponivel: R$"+saldo);

            try {
                System.out.print("Digite o valor que deseja sacar: ");
                double valorSaque = sc.nextDouble();

                if (valorSaque<=0) {
                    System.out.println("Erro: O valor do saque nao pode ser negativo e 0.");
                }
                else if (valorSaque>saldo) {
                    System.out.println("Erro: Saldo insuficiente.");
                }
                else {
                    saldo-=valorSaque;
                    System.out.println("Saque realizado com sucesso...");
                    System.out.printf("Novo saldo: R$%.2f%n",saldo);
                }
            }catch (InputMismatchException e) {
                System.out.println("Erro Critico: Entrada invalida!! Por favor, use apenas numeros e virgula");
            }catch (Exception e) {
                System.out.println("Erro Inesperado: "+e.getMessage());
            } finally {
                System.out.println("Operacao Finalizada");
            }


        sc.close();
    }
}
